test/test_prerun.py

```diff
@@ -11,35 +13,113 @@
-def test_prerun_reqs_v1() -> None:
+@pytest.mark.parametrize(
+    ("deactivate"),
+    (
+        (False),
+        (True),
+    ),
+    ids=("venv activated", "venv deactivated"),
+)
+def test_prerun_reqs_v1(deactivate: bool, deactivate_venv: Deactivator) -> None:
     """Checks that the linter can auto-install requirements v1 when found."""
     cwd = os.path.realpath(
         os.path.join(
             os.path.dirname(os.path.realpath(__file__)), "..", "examples", "reqs_v1"
         )
     )
+    execdir = deactivate_venv(deactivate)
     result = run_ansible_lint("-v", ".", cwd=cwd)
-    assert "Running ansible-galaxy role install" in result.stderr, result.stderr
+
+    ansible_galaxy = os.path.join(execdir, "ansible-galaxy")
+    assert f"Running {ansible_galaxy} role install" in result.stderr, result.stderr
     assert (
-        "Running ansible-galaxy collection install" not in result.stderr
+        f"Running {ansible_galaxy} collection install" not in result.stderr
     ), result.stderr
     assert result.returncode == 0, result
 
 
 @flaky(max_runs=3)  # type: ignore
-def test_prerun_reqs_v2() -> None:
+@pytest.mark.parametrize(
+    ("executable"),  # Run with `python -m` or the venv command: issue #1507
+    (
+        (None),
+        (os.path.join(os.path.dirname(sys.executable), "ansible-lint")),
+    ),
+    ids=(
+        "run as ansiblelint python module",
+        "run via path to the virtualenv's ansible-lint command",
+    ),
+)
+@pytest.mark.parametrize(
+    ("deactivate"),
+    (
+        (False),
+        (True),
+    ),
+    ids=("venv activated", "venv deactivated"),
+)
+def test_prerun_reqs_v2(
+    executable: str, deactivate: bool, deactivate_venv: Deactivator
+) -> None:
     """Checks that the linter can auto-install requirements v2 when found."""
     cwd = os.path.realpath(
         os.path.join(
             os.path.dirname(os.path.realpath(__file__)), "..", "examples", "reqs_v2"
         )
     )
-    result = run_ansible_lint("-v", ".", cwd=cwd)
-    assert "Running ansible-galaxy role install" in result.stderr, result.stderr
-    assert "Running ansible-galaxy collection install" in result.stderr, result.stderr
+    execdir = deactivate_venv(deactivate)
+    result = run_ansible_lint("-v", ".", cwd=cwd, executable=executable)
+
+    # These first 2 assertations are written to pass both with the test
+    # as-it-is before the fix of issue #1507 and after the fix.
+    # (See: https://github.com/ansible-community/ansible-lint/issues/1507)
+    # But these assertations are written as much like the "before" version
+    # of this test as possible, to demonstrate that the assertion
+    # itself is not substantively changed by the fix.
+    # When these asserations fail, without the #1507 patch,
+    # it demonstrates that ansible-lint does not run in a virtual environment
+    # unless that environment is activated.
+    assert re.search(
+        "Running [^ ]*ansible-galaxy role install", result.stderr
```

Comment:
Why not do a more precise comparison with
```suggestion
        f"Running {execdir}/ansible-galaxy role install", result.stderr
```
?